### DISCUSSION TITLE
개발 능률 올리기

### DISCUSSION THREAD

**Discussion by airman5573** [woowacourse-study/2022-woowahan-bansanghwe#151] [ETC]
# 결론
1. 마인드맵을 사용하자
2. 문제 정의를 제대로
3. ChatGPT를 활용하자
4. Stackoverflow를 활용하자

# 마인드맵을 사용하자
마인드맵은 말그대로 마음의 지도입니다.
그리고 지도는 현재 내가 어디에 있고 목적지는 어디에 있는지, 그 목적지까지 빠르게 가기 위해서는 어떤 길로 가야 하는지 알려줍니다. 
저는 마인드맵을 아래의 경우에 활용합니다.
1. 아티클을 읽을때
2. 복잡한 소스 코드를 분석할때
3. 머리가 복잡해서 그냥 아무거나 적을때

## 마인드맵 + 아티클
아티클을 읽을때 목차를 마인드맵으로 정리하고, 해당 단원에서 핵심적인 내용을 기록합니다.
![sc_ 48](https://user-images.githubusercontent.com/68623798/215453509-7556d277-ebc8-495f-8c41-08e48c9a77e5.png)
이렇게 정리해 놓으면 복습하기도 쉽고, 더 능동적으로 읽게 됩니다.
이를테면 읽다가 떠오르는 질문들을 쉽게 적을 수 있습니다.
![sc_ 49](https://user-images.githubusercontent.com/68623798/215456523-9e73fb10-3979-47c7-b5d4-a9f5fbfdd7b8.png)

## 복잡한 소스 코드를 분석할때
다른 사람이 짠 코드를 읽을때 마인드맵을 적극 활용합니다.
![sc_ 50](https://user-images.githubusercontent.com/68623798/215456842-4b8f27cb-b926-4f1a-b13d-ad0a024bfbfa.png)

코드 호출을 타고타고 가다보면 길을 잃기 마련입니다. '잉? 어쩌다 여기까지 왔지?'  
마인드맵으로 읽고 있는 코드를 기록해 놓으면 길을 잃지 않습니다.

## 아무거나 적기
어떤 문제를 풀때 그 해결 과정이 머릿속에서 착착착착 생각하기란 쉽지 않습니다.  
머릿속에서 다양한 시뮬레이션을 돌려보면서 답으로 가는 적당한 길을 찾는데,  
이때 역시 길을 잃기 쉽습니다. 좋은 아이디어도 머릿속에서 등장했다가 잠깐 딴생각하면 사라집니다.  (저만 그럴지도..?)  
그래서 저는 일단 그럴싸한 아이디어가 생각나면 적습니다.
![sc_ 51](https://user-images.githubusercontent.com/68623798/215457969-6fd553e8-d289-448c-bfc2-3b72b93abd4e.png)

이렇게 적어놓고, 하나하나 실험해봅니다.

# 문제 정의를 제대로
요즘 오류수정 알바를 하는데, 문제 정의의 중요성을 많이 느낍니다.  
열심히 개발했는데, 사실 고객이 원하는것과 다른 결과물을 만든것입니다.  
현타오고 그때 고객에게 질문을 더 꼼꼼히 해서 문제를 제대로 정의할껄,,하는 후회를 합니다.  

문제 정의를 잘하기 위해서 '**질문**'을 많이 하면 된다고 생각합니다.  
'에이 이정도는 내 예상이 맞겠지...' 라고 넘겨 짚기보다는,
그래도 한번 더 질문해서 무엇을 원하는지, 뭐가 문제인지 명확히 하는것이 안전합니다.

# ChatGPT를 활용하자
ChatGPT는 센세이션입니다. 저는 구글링보다 이제는 ChatGPT에게 더 많이 물어봅니다.  
물론 자주 잘못된 정보를 주지만, 꼬치꼬치 캐물으면 그래도 정답에 가까운 답변을 줍니다.  
정말 잘못된 답변이더라도 해당 키워드로 구글링하면 빠르게 원하는 정보를 찾을 수 있습니다.
최근에 슬슬 유료버전도 제공하려는것 같습니다.  
사전예약(?)은 [여기서](https://docs.google.com/forms/d/e/1FAIpQLSfCVqahRmA5OxQXbRlnSm531fTd8QBdUCwZag7mI9mrlOOIaw/viewform) 가능합니다.

# Stackoverflow를 활용하자
딱 봐도 어려운 문제들이 있습니다.  
감각적으로 오래 걸릴것 같은 문제들은 Stackoverflow에 먼저 질문을 올려놓고  
그 다음에 삽질을 하면 효율적입니다.  

무엇이 문제인지 질문을 작성하면서 스스로 문제를 명확하게 알 수 있고  
양키 고수 형님들이 친절하게 알려줍니다.  

이상입니다. 읽어주셔서 감사합니다.

**Comment by JUDONGHYEOK** [comment]
이렇게하면 병민처럼 될 수 있나요?

**Reply by airman5573** [reply]
저처럼 되지 마쇼 ㅋㅋ
